tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -238,6 +242,59 @@ void diagnostics_test() {
     }
 }
 
+template <template <class...> class TupleLike>
+constexpr bool tuple_like_test() {
+    {
+        constexpr TupleLike<int, int> t1{1, 1};
+        constexpr TupleLike<int, int> t1_equal{1, 1};
+        constexpr TupleLike<int, int> t2{2, 1};
+
+        static_assert(spaceship_test<std::strong_ordering>(t1, t1_equal, t2));
+    }
+    {
+        constexpr TupleLike<int, double> t1{1, 1.0};
+        constexpr TupleLike<int, double> t1_equal{1, 1.0};
+        constexpr TupleLike<int, double> t2{2, 1.0};
+
+        static_assert(spaceship_test<std::partial_ordering>(t1, t1_equal, t2));
+    }
+
+    return true;
+}
+
+template <auto SmallVal, decltype(SmallVal) EqualVal, decltype(EqualVal) LargeVal>
+bool constexpr optional_test() {
+    using ReturnType = std::compare_three_way_result_t<decltype(SmallVal), decltype(LargeVal)>;

```

Comment:
`decltype(LargeVal)` can't vary (it's always the same as `SmallVal`), so we can simplify this to the 1-arg form.